### ISSUE TITLE
fix: set spinbutton role in NumberField

### ISSUE THREAD

**Issue by m-akinc** [microsoft/fast#6148]
# 🐛 Bug Report

The FAST foundation number field has [spinbutton](https://www.w3.org/WAI/ARIA/apg/patterns/spinbutton/) semantics, so the input control should set the `role` attribute to "spinbutton".

## 💻 Repro or Code Sample

Examine the accessibility tree in dev tools. You'll see the number field is listed as "textbox" rather than "spinbutton".

## 🤔 Expected Behavior

Accessibility role should be "spinbutton".

## 😯 Current Behavior

Role is not specified, so accessibility tree shows item as "textbox".

## 💁 Possible Solution

Set role on `input` element to "spinbutton".

## 🔦 Context

I am trying to conform to accessibility standards.

## 🌍 Your Environment

N/A

**Comment by chrisdholt** [comment]
Can you elaborate on this:
> The FAST foundation number field has [spinbutton](https://www.w3.org/WAI/ARIA/apg/patterns/spinbutton/) semantics, so the input control should set the role attribute to "spinbutton".

I'm not entirely sure that spinbutton is correct here as the intent here is to essentially provide encapsulation to a input with type of number, essentially a "number" field in a form. This will need more investigation and any additional information you have around the above or investigation that you've done will be helpful.

**Comment by m-akinc** [comment]
> Can you elaborate on this:
> 
> > The FAST foundation number field has [spinbutton](https://www.w3.org/WAI/ARIA/apg/patterns/spinbutton/) semantics, so the input control should set the role attribute to "spinbutton".
> 
> I'm not entirely sure that spinbutton is correct here as the intent here is to essentially provide encapsulation to a input with type of number, essentially a "number" field in a form. This will need more investigation and any additional information you have around the above or investigation that you've done will be helpful.

The FAST number field has a text input control and "step-up" and "step-down" buttons, where up and down arrow on the keyboard increment/decrement the value. The component also supports `min` and `max` attributes to limit the value range. The description of a spinbutton (on the page I linked) matches this pretty closely: 

> A spinbutton is an input widget that restricts its value to a set or range of discrete values. For example, in a widget that enables users to set an alarm, a spinbutton could allow users to select a number from 0 to 59 for the minute of an hour.
>
>Spinbuttons often have three components, including a text field that displays the current value, an increase button, and a decrease button. The text field is usually the only focusable component because the increase and decrease functions are keyboard accessible via arrow keys. Typically, the text field also allows users to directly edit the value.

Granted, the number field isn't _necessarily_ limited to a finite set of values, but I'm not sure that disqualifies it, given how well it matches the rest of the criteria.

**Comment by rajsite** [comment]
[`spinbutton` is also the implicit role for `<input type="number">`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#accessibility) which it seems like number-field is an analog for.

**Comment by janechu** [comment]
Unfortunately `@microsoft/fast-foundation` is being deprecated, refer to #6955.
